source-code/java/src/isse/mbr/parsing/CodeGenerator.java

```diff
@@ -631,29 +631,29 @@ private void addMiniSearchPredicates(PVSInstance inst, StringBuilder sb, MiniBra
 	private void addDecisionVariables(PVSInstance inst, StringBuilder sb, MiniBrassAST model, Map<String, String> subs, String overallIdent, String valuationsArray, StringBuilder instanceArguments) {
 		sb.append("\n% Decision variables: \n");
 		PVSType pvsType = inst.getType().instance;
-		
+
 		appendOverall(sb, pvsType.getElementType(), inst, overallIdent);
 
 		PVSFormalParameter numberSoftConstraints = pvsType.getParamMap().get(PVSType.N_SCS_LIT);
 		IntervalType softConstraintsSet = new IntervalType(new NumericValue(1), new NumericValue(numberSoftConstraints));
 
-		
+
 		MiniZincParType specType = pvsType.getSpecType();
-		if (specType instanceof MiniZincVarType) {
-			sb.append(String.format("array[%s] of var %s: %s;\n", softConstraintsSet.toMzn(inst),
-					encode(pvsType.getSpecType(), inst), valuationsArray));
-		} else {
+		if (specType instanceof MiniZincArrayLike) {
 			MiniZincArrayLike arrayLike = (MiniZincArrayLike) specType;
 			ArrayType arrayType = arrayLike.getArrayType();
 			sb.append(String.format("array[%s,%s] of var %s: %s;\n", softConstraintsSet.toMzn(inst),
 					getArrayIndexSets(arrayType, inst), encode(arrayType.getElementType(), inst), valuationsArray));
+		} else {
+			sb.append(String.format("array[%s] of var %s: %s;\n", softConstraintsSet.toMzn(inst),
+					encode(pvsType.getSpecType(), inst), valuationsArray));
 		}
```

Comment:
Swapped if/else. This should not really change anything, but we we now firstly have a safe typecast to `MiniZincArrayLike` and secondly use the same order as in the soft constraint generation, ensuring that both work well together.